MekHQ/src/mekhq/gui/InterstellarMapPanel.java

```diff
@@ -738,6 +738,16 @@ protected void paintComponent(Graphics g) {
 	                        if(null != factions && !isSystemEmpty(system)) {
 	                            int i = 0;
 	                            for(Faction faction : factions) {
+	                            	if (faction.getStartingPlanet(campaign.getGameYear()).equals(system.getId())) {
```

Comment:
Hoist `faction.getStartingPlanet(campaign.getGameYear())` into a local variable outside the for loop.